compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/JvmOptimizationLowering.kt

```diff
@@ -63,6 +65,41 @@ class JvmOptimizationLowering(val context: JvmBackendContext) : FileLoweringPass
             else -> null
         }
 
+    private fun IrExpression.coerceTo(target: IrType): IrExpression =
+        IrCallImpl(
+            startOffset,
+            endOffset,
+            target,
+            context.ir.symbols.unsafeCoerceIntrinsic
+        ).also { call ->
+            call.putTypeArgument(0, type)
+            call.putTypeArgument(1, target)
+            call.putValueArgument(0, this)
+        }
+
+    private fun IrCall.replaceWithCallTo(replacement: IrFunctionSymbol) =
```

Comment:
To be clear, the calls being replaced are all static function calls, right? So no need to copy receivers?

Reply:
That's right! Should I make that more clear?